lib/handler/proxy.c

```diff
@@ -59,6 +59,13 @@ static int on_req(h2o_handler_t *_self, h2o_req_t *req)
         authority = &self->upstream.authority;
     }
 
+    /* rewrite headers */
+    if (self->config.header_cmds.size != 0) {
+        h2o_headers_command_t *cmd;
+        for (cmd = self->config.header_cmds.entries; cmd->cmd != H2O_HEADERS_CMD_NULL; ++cmd)
+            h2o_rewrite_headers(&req->pool, &req->headers, cmd);
+    }
```

Comment:
Could we just set the list of header modification commands to `req->overrides` and apply them only when the request is sent upstream in lib/core/proxy.c?

The reason I ask is because an upstream server (accessed using the modified headers) might return 399. In such case, the request would be delegated to the next handler. I think that the headers of the original request should be passed to the next handler, since per my understanding the intended behavior of `proxy.header.*` is to modify the headers passed to the upstream server only.